### ISSUE TITLE
Sub-subfolders not accesible on public share links with password set when Brute force protection is on

### ISSUE THREAD

**Issue by brozkeff** [owncloud/core#38108]
### Steps to reproduce
1. Upgrade ownCloud server to 10.5 from 10.4 when you have Brute force protection plugin on
2. Pick any public sharing link with already set password, or create a new sharing link WITH A PASSWORD, with structure containing more than two levels of folders
3. Try to access sub-sub-folders inside this sharink link with password. Error on top "This folder is not available, check logs or contact your admin" (Czech: "Tento adresář není dostupný, zkontrolujte prosím logy nebo kontaktujte svého správce systému") and sub-sub-folders cannot be accessed. Removing password from the sharing link restores accessibility also of sub-subfolders. Downloading of entire sharing link contents via button on top right in ZIP works and is the only known workaround to download contents otherwise inaccessible by UI.

### Expected behaviour
Password-protected sharing links should work including any level of sub-folders as before even with Brute force protection plugin on. Before upgrade the plugin did not cause any bad behaviour.

### Actual behaviour
Sub-subfolders cannot be accessed due to error "This folder is not available, check logs or contact your admin" (Czech: "Tento adresář není dostupný, zkontrolujte prosím logy nebo kontaktujte svého správce systému"). After discovering the trigger from the plugin Brute force Protection (v1.1.0 by Semih Serhat Karakaya) and disabling them plugin password-protected links work OK.

### Server configuration
**Operating system**: Ubuntu 18.04

**Web server:** Apache/2.4.29 (Ubuntu)

**Database:** mysql  Ver 14.14 Distrib 5.7.32

**PHP version:** PHP 7.2.34-8+ubuntu18.04.1+deb.sury.org+1

**ownCloud version:** 10.5.0 (stable) 

**Updated from an older ownCloud or fresh install:** update from 10.4

**Where did you install ownCloud from:** apt repository

**Signing status (ownCloud 9.0 and above):**

```
No errors have been found.
```

**The content of config/config.php:**

```
{
    "system": {
        "installed": true,
        "dbtype": "mysql",
        "dbname": "oc",
        "dbuser": "***REMOVED SENSITIVE VALUE***",
        "dbpassword": "***REMOVED SENSITIVE VALUE***",
        "dbhost": "localhost",
        "dbtableprefix": "oc_",
        "passwordsalt": "***REMOVED SENSITIVE VALUE***",
        "forcessl": true,
        "blacklisted_files": [
            ".htaccess"
        ],
        "overwritehost": "",
        "overwriteprotocol": "",
        "overwritewebroot": "",
        "overwritecondaddr": "",
        "trusted_domains": [
            "babylon.tierraverde.cz"
        ],
        "theme": "",
        "default_language": "cs_CZ",
        "3rdpartyroot": "",
        "3rdpartyurl": "",
        "defaultapp": "files",
        "knowledgebaseenabled": true,
        "mail_domain": "***REMOVED SENSITIVE VALUE***",
        "mail_smtpdebug": false,
        "mail_smtpmode": "smtp",
        "mail_smtphost": "***REMOVED SENSITIVE VALUE***",
        "mail_smtpport": "587",
        "mail_smtptimeout": 10,
        "mail_smtpsecure": "tls",
        "mail_smtpauth": true,
        "mail_smtpauthtype": "LOGIN",
        "mail_smtpname": "***REMOVED SENSITIVE VALUE***",
        "mail_smtppassword": "***REMOVED SENSITIVE VALUE***",
        "filelocking.enabled": "true",
        "memcache.local": "\\OC\\Memcache\\Redis",
        "memcache.locking": "\\OC\\Memcache\\Redis",
        "redis": {
            "host": "\/var\/run\/redis\/redis.sock",
            "port": 0,
            "timeout": 0
        },
        "trashbin_retention_obligation": "30, auto",
        "allow_user_to_change_display_name": true,
        "appcodechecker": false,
        "updatechecker": false,
        "has_internet_connection": true,
        "check_for_working_webdav": true,
        "check_for_working_htaccess": true,
        "log_type": "owncloud",
        "logfile": "",
        "loglevel": 3,
        "logdateformat": "F d, Y H:i:s",
        "logtimezone": "Europe\/Prague",
        "log_query": false,
        "log_authfailip": true,
        "cron_log": true,
        "log_rotate_size": 10485760,
        "log_rotate_sizerotation": 2147483648,
        "remember_login_cookie_lifetime": 1296000,
        "session_lifetime": 86400,
        "session_keepalive": true,
        "custom_csp_policy": "default-src 'self'; script-src 'self' 'unsafe-eval'; style-src 'self' 'unsafe-inline'; frame-src *; img-src *; font-src 'self' data:; media-src *",
        "xframe_restriction": true,
        "maintenance": false,
        "apps_paths": [
            {
                "path": "\/var\/www\/owncloud\/apps",
                "url": "\/apps",
                "writable": true
            }
        ],
        "customclient_desktop": "http:\/\/owncloud.org\/sync-clients\/",
        "customclient_android": "https:\/\/f-droid.org\/repository\/browse\/?fdid=com.owncloud.android",
        "customclient_ios": "https:\/\/itunes.apple.com\/us\/app\/owncloud\/id543672169?mt=8",
        "enable_previews": true,
        "preview_max_scale_factor": 1,
        "enabledPreviewProviders": {
            "0": "OC\\Preview\\PNG",
            "1": "OC\\Preview\\JPEG",
            "2": "OC\\Preview\\GIF",
            "11": "OC\\Preview\\Illustrator",
            "12": "OC\\Preview\\Postscript",
            "13": "OC\\Preview\\Photoshop",
            "14": "OC\\Preview\\TIFF",
            "15": "OC\\Preview\\MSOfficeDoc",
            "16": "OC\\Preview\\OpenDocument",
            "17": "OC\\Preview\\PDF",
            "18": "OC\\Preview\\MSOffice2003",
            "19": "OC\\Preview\\MSOffice2007",
            "20": "OC\\Preview\\SVG",
            "21": "OC\\Preview\\TIFF"
        },
        "preview_libreoffice_path": "\/usr\/bin\/libreoffice",
        "preview_office_cl_parameters": "",
        "enable_avatars": true,
        "openssl": [],
        "singleuser": false,
        "instanceid": "oc1fdcd11858",
        "datadirectory": "\/srv\/data",
        "version": "10.5.0.10",
        "mail_from_address": "***REMOVED SENSITIVE VALUE***",
        "secret": "***REMOVED SENSITIVE VALUE***",
        "minimum.supported.desktop.version": "2.4.0",
        "integrity.ignore.missing.app.signature": [
            "XXXXXtheme"
        ]
    }
}

```

**List of activated apps:**

```
Enabled:
  - activity: 2.6.0
  - announcementcenter: 1.2.1
  - brute_force_protection: 1.1.0
  - calendar: 1.6.4
  - camerarawpreviews: 0.7.8
  - comments: 0.3.0
  - configreport: 0.2.0
  - contacts: 1.5.5
  - customgroups: 0.6.0
  - dav: 0.6.0
  - federatedfilesharing: 0.5.0
  - federation: 0.1.0
  - files: 1.5.2
  - files_external: 0.7.1
  - files_mediaviewer: 1.0.3
  - files_pdfviewer: 0.11.2
  - files_sharing: 0.13.0
  - files_texteditor: 2.3.0
  - files_trashbin: 0.9.1
  - files_versions: 1.3.0
  - gallery: 16.1.1
  - impersonate: 0.5.0
  - market: 0.6.0
  - notifications: 0.5.2
  - provisioning_api: 0.5.0
  - systemtags: 0.3.0
  - XXXtheme: 1.0.0
  - twofactor_totp: 0.7.0
  - updatenotification: 0.2.1
Disabled:
  - drawio
  - encryption
  - external
  - firstrunwizard
  - oauth2
  - user_external
  - wallpaper
```

**Are you using external storage, if yes which one:** none

**Are you using encryption:** no

**Are you using an external user-backend, if yes which one:** none

### Client configuration
**Browser:** Firefox 82.0.2

**Operating system:** Linux Mint 18.3 64bit

#### ownCloud log (data/owncloud.log)
```
{"reqId":"XXX","level":4,"time":"November 12, 2020 14:21:54","remoteAddr":"xxx","user":"--","app":"webdav","method":"PROPFIND","url":"\/public.php\/webdav\/EN\/doc","message":"Exception: Too many failed attempts. Try again in  5 minutes.: {\"Exception\":\"OCA\\\\BruteForceProtection\\\\E
xceptions\\\\LinkAuthException\",\"Message\":\"Too many failed attempts. Try again in  5 minutes.\",\"Code\":0,\"Trace\":\"#0 \\\/var\\\/www\\\/owncloud\\\/apps\\\/brute_force_protection\\\/lib\\\/Hooks.php(159): OCA\\\\BruteForceProtection\\\\Throttle->applyBruteForcePolicyForLinkShare('gjz330bG0I9i6d9', '83.240
.7.150')\\n#1 \\\/var\\\/www\\\/owncloud\\\/lib\\\/composer\\\/symfony\\\/event-dispatcher\\\/EventDispatcher.php(264): OCA\\\\BruteForceProtection\\\\Hooks->preLinkShareAuthCallback(Object(Symfony\\\\Component\\\\EventDispatcher\\\\GenericEvent), 'share.beforepas...', Object(Symfony\\\\Component\\\\EventDispatch
er\\\\EventDispatcher))\\n#2 \\\/var\\\/www\\\/owncloud\\\/lib\\\/composer\\\/symfony\\\/event-dispatcher\\\/EventDispatcher.php(239): Symfony\\\\Component\\\\EventDispatcher\\\\EventDispatcher->doDispatch(Array, 'share.beforepas...', Object(Symfony\\\\Component\\\\EventDispatcher\\\\GenericEvent))\\n#3 \\\/var\\
\/www\\\/owncloud\\\/lib\\\/composer\\\/symfony\\\/event-dispatcher\\\/EventDispatcher.php(73): Symfony\\\\Component\\\\EventDispatcher\\\\EventDispatcher->callListeners(Array, 'share.beforepas...', Object(Symfony\\\\Component\\\\EventDispatcher\\\\GenericEvent))\\n#4 \\\/var\\\/www\\\/owncloud\\\/lib\\\/private\
\\/Share20\\\/Manager.php(1496): Symfony\\\\Component\\\\EventDispatcher\\\\EventDispatcher->dispatch(Object(Symfony\\\\Component\\\\EventDispatcher\\\\GenericEvent), Object(Symfony\\\\Component\\\\EventDispatcher\\\\GenericEvent))\\n#5 \\\/var\\\/www\\\/owncloud\\\/apps\\\/dav\\\/lib\\\/Connector\\\/PublicAuth.p
hp(98): OC\\\\Share20\\\\Manager->checkPassword(*** sensitive parameters replaced ***)\\n#6 \\\/var\\\/www\\\/owncloud\\\/lib\\\/composer\\\/sabre\\\/dav\\\/lib\\\/DAV\\\/Auth\\\/Backend\\\/AbstractBasic.php(103): OCA\\\\DAV\\\\Connector\\\\PublicAuth->validateUserPass(*** sensitive parameters replaced ***)\\n#7
\\\/var\\\/www\\\/owncloud\\\/lib\\\/composer\\\/sabre\\\/dav\\\/lib\\\/DAV\\\/Auth\\\/Plugin.php(183): Sabre\\\\DAV\\\\Auth\\\\Backend\\\\AbstractBasic->check(Object(Sabre\\\\HTTP\\\\Request), Object(Sabre\\\\HTTP\\\\Response))\\n#8 \\\/var\\\/www\\\/owncloud\\\/lib\\\/composer\\\/sabre\\\/dav\\\/lib\\\/DAV\\\/A
uth\\\/Plugin.php(137): Sabre\\\\DAV\\\\Auth\\\\Plugin->check(Object(Sabre\\\\HTTP\\\\Request), Object(Sabre\\\\HTTP\\\\Response))\\n#9 \\\/var\\\/www\\\/owncloud\\\/lib\\\/composer\\\/sabre\\\/event\\\/lib\\\/WildcardEmitterTrait.php(89): Sabre\\\\DAV\\\\Auth\\\\Plugin->beforeMethod(Object(Sabre\\\\HTTP\\\\Reque
st), Object(Sabre\\\\HTTP\\\\Response))\\n#10 \\\/var\\\/www\\\/owncloud\\\/lib\\\/composer\\\/sabre\\\/dav\\\/lib\\\/DAV\\\/Server.php(454): Sabre\\\\DAV\\\\Server->emit('beforeMethod:PR...', Array)\\n#11 \\\/var\\\/www\\\/owncloud\\\/lib\\\/composer\\\/sabre\\\/dav\\\/lib\\\/DAV\\\/Server.php(251): Sabre\\\\DAV
\\\\Server->invokeMethod(Object(Sabre\\\\HTTP\\\\Request), Object(Sabre\\\\HTTP\\\\Response))\\n#12 \\\/var\\\/www\\\/owncloud\\\/lib\\\/composer\\\/sabre\\\/dav\\\/lib\\\/DAV\\\/Server.php(319): Sabre\\\\DAV\\\\Server->start()\\n#13 \\\/var\\\/www\\\/owncloud\\\/apps\\\/dav\\\/appinfo\\\/v1\\\/publicwebdav.php(1
05): Sabre\\\\DAV\\\\Server->exec()\\n#14 \\\/var\\\/www\\\/owncloud\\\/public.php(75): require_once('\\\/var\\\/www\\\/ownclo...')\\n#15 {main}\",\"File\":\"\\\/var\\\/www\\\/owncloud\\\/apps\\\/brute_force_protection\\\/lib\\\/Throttle.php\",\"Line\":112}"}
```

Basically from the owncloud log it appears as if some code change in ownCloud 10.5 causes Brute Force protection plugin to reject access to sub-sub-folders of password-protected shares. Workaround is disabling Brute force protection whis is not optimal. Restoring db from backup and downgrading to 10.4 is neither desired.

**Comment by karakayasemi** [comment]
This bug identified in https://github.com/owncloud/brute_force_protection/issues/138 and resolved with https://github.com/owncloud/core/pull/38016 .

It will work okay with next release of owncloud server 10.6. For 10.5, you can apply the patch in https://github.com/owncloud/core/pull/38016. Thank you for reporting.
